sql/analyzer/fix_exec_indexes.go

```diff
@@ -432,10 +432,15 @@ func fixExprToScope(e sql.Expression, scopes ...*idxScope) sql.Expression {
 	ret, _, _ := transform.Expr(e, func(e sql.Expression) (sql.Expression, transform.TreeIdentity, error) {
 		switch e := e.(type) {
 		case *expression.GetField:
-			// TODO: this is a swallowed error. It triggers falsely in queries involving the dual table, or queries where
-			//  the columns being selected are only found in subqueries
+			// TODO: this is a swallowed error in some cases. It triggers falsely in queries involving the dual table, or
+			//  queries where the columns being selected are only found in subqueries. Conversely, we actually want to ignore
+			//  this error for the case of DEFAULT in a `plan.Values`, since we analyze the insert source in isolation (we
```

Comment:
i have been trying to untangle this for awhile, if you have any good ideas for how to one-and-done insert indexing

Reply:
maybe new node types might help? `*plan.InsertValues` and others to differentiate? not in-scope ofc

Reply:
Probably possible to do upfront indexing in the cases where this applies, then have a Node interface extension that allows them to opt out of this indexing step? Then this can correctly error in the remaining cases, rather than at runtime.